### ISSUE TITLE
Problem: Enduro Ingest workflows are not easily extensible

### ISSUE THREAD

**Issue by fiver-watson** [artefactual-sdps/enduro#886]
Enduro Ingest aims to help the keepers of cultural memory preserve their content. However, Enduro is also an open source project trying to reach an international user base, which spreads across many different jurisdictions, uses different national and regional standards, follows many varying local policies, are trying to address many different use cases, and correspondingly can have many different submission requirements from upstream content creators.

At present, a3m and Archivematica  - the underlying preservation engines of the system - accept a very limited number of variations in the expected incoming SIP structure. These limitations do not align with the many variations in Transfer structures that might be submitted by Enduro users and upstream creators. 

Currently there is no established method of customizing and/or extending Enduro's functionality, short of forking the project. This makes the ongoing maintenance of production Enduro instances across different users and jurisdictions difficult. 

In short - There is no well-defined and project-supported method to extend default workflows to account for specific user needs, particularly in regards to pre-preservation processing activities. Enduro would greatly benefit from a standardized and documented method of extending the existing workflows locally.

**Comment by sallain** [comment]
Still needs documentation and a little more work but this can be closed in general
